### PR TITLE
feat(vision): add GCP Vision API integration

### PR DESCRIPTION
**Pull request by Trinath052004**
## Description

Added Google Cloud Vision API integration with 9 detection capabilities including label detection, OCR, face detection, object localization, and more

## Type of Change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Refactoring (no functional changes)

## Related Issues 

Closes #3616

## Changes Made

- Added `vision_tool` module with 9 Google Cloud Vision API detection functions
- Updated `integrations.py` to support GCP Vision API credentials
- Added comprehensive error handling for file validation and API errors
- Created 24 unit tests covering all features and edge cases
- Added README documentation with usage examples

## Testing

Describe the tests you ran to verify your changes:

- [x] Unit tests pass (`cd core && pytest tests/`)
- [x] Lint passes (`cd core && ruff check .`)
- [x] Manual testing performed

## Checklist

- [x] My code follows the project's style guidelines
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Screenshots (if applicable)

N/A -Backend tool integration

### CODE REVIEW

File: tools/src/aden_tools/credentials/integrations.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        ],
+        required=True,
+        startup_required=False,
+        help_url="https://console.cloud.google.com/apis/credentials",
+        description="Google Cloud Vision API key for image analysis",
+        # Auth method support
+        aden_supported=False,
+        aden_provider_name="",
+        direct_api_key_supported=True,
+        api_key_instructions="""To get a Google Cloud Vision API key:
```
Comment (prakharsingh-74) [inline tools/src/aden_tools/credentials/integrations.py]: add API key instructions in readme file instead of here

**Review by prakharsingh-74 — CHANGES_REQUESTED** [review]

**Review by prakharsingh-74 — CHANGES_REQUESTED** [review]
please add all the reference screenshot also, that this issue has been resolved

File: tools/src/aden_tools/credentials/integrations.py (hunk 1 above)
Comment (Trinath052004) [reply tools/src/aden_tools/credentials/integrations.py]: Thanks for suggestion!, I noticed @RichardTang-Aden deleted the integrations.py and added .py files to each tool, following the same pattern-I created gcp_vision.py file .
Hope, this will resolve ur  concerns.
ThankYou.

**Comment by Trinath052004** [comment]
Initially added the Vision integration credential to older integration.py structure.
After syncing with latest main, I aligned with the updated architecture and moved the logic to the gcp_vision.py; removing the old file.
Happy to make further  changes :)

File: tools/src/aden_tools/credentials/__init__.py [hunk 2]
```diff
@@ -73,6 +77,8 @@
     **LLM_CREDENTIALS,
     **SEARCH_CREDENTIALS,
     **EMAIL_CREDENTIALS,
+    **INTEGRATION_CREDENTIALS,
```
Comment (Hundao) [inline tools/src/aden_tools/credentials/__init__.py]: `INTEGRATION_CREDENTIALS` is never imported or defined anywhere. This will crash at runtime.

File: tools/src/aden_tools/credentials/__init__.py [hunk 3]
```diff
@@ -50,6 +50,10 @@
 3. If new category, import and merge it in this __init__.py
 """
 
+from .base import CredentialError, CredentialSpec
+from .browser import get_aden_auth_url, get_aden_setup_url, open_browser
```
Comment (Hundao) [inline tools/src/aden_tools/credentials/__init__.py]: These imports already exist a few lines below, remove the duplicates.

File: .gitignore [hunk 4]
```diff
@@ -74,4 +74,4 @@ exports/*
 
 docs/github-issues/*
 core/tests/*dumps/*
-screenshots/*
\ No newline at end of file
+agent_logs/
```
Comment (Hundao) [inline .gitignore]: This is unrelated to the Vision tool, please revert.

**Review by Hundao — CHANGES_REQUESTED** [review]
Left some comments inline. Can you also share some e2e test output showing a couple tools working? Thanks!

**Review by prakharsingh-74 — CHANGES_REQUESTED** [review]
Hey @Trinath052004 please check the lint test they are failing you can check it out here - https://github.com/adenhq/hive/actions/runs/21870639828/job/63208638027?pr=4231

Also please do the changes suggested by @Hundao

**Comment by Trinath052004** [comment]
> Left some comments inline. Can you also share some e2e test output showing a couple tools working? Thanks!

Hi @Hundao i have made suggested changes and ran e2e tests, the results are as follows:

**Test 1:**
Screenshot 2026-02-11 201825

**Test2:**
Screenshot 2026-02-11 203619

btw , i apologize for creating quite a number of commits( i jus wanted to make sure everything works perfect!)



> Hey @Trinath052004 please check the lint test they are failing you can check it out here - https://github.com/adenhq/hive/actions/runs/21870639828/job/63208638027?pr=4231
> 
> Also please do the changes suggested by @Hundao

Thankyou @prakharsingh-74 for the review and suggestion( i have updated accordingly!)

--------------------------------------------------------------------------------------------------------------------------------------------

@Hundao , i am open to make further changes if necessary, Thankyou for trusting me by assigning this issue to me :)

File: tools/src/aden_tools/credentials/gcp_vision.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+        ],
+        required=True,
+        startup_required=False,
+        help_url="https://console.cloud.google.com/apis/credentials",
+        description="Google Cloud Vision API key for image analysis",
+        # Auth method support
+        aden_supported=False,
+        aden_provider_name="",
+        direct_api_key_supported=True,
+        api_key_instructions="""To get a Google Cloud Vision API key:
```
Comment (prakharsingh-74) [inline tools/src/aden_tools/credentials/gcp_vision.py]: Is it necessary to write instructions here? 
you can add them seperately in .md file also

File: .gitignore [hunk 6]
```diff
@@ -74,4 +74,6 @@ exports/*
 
 docs/github-issues/*
 core/tests/*dumps/*
-screenshots/*
\ No newline at end of file
+
```
Comment (prakharsingh-74) [inline .gitignore:77]: try to do minimum changes in the file, dont do the changes in the file by just reducing their spaces

**Review by prakharsingh-74 — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 6 above)
Comment (Trinath052004) [reply .gitignore:77]: I removed agentlogs and added screenshots/* as suggested by @hundao ,not jus reducing spaces

Comment (Trinath052004) [reply .gitignore:77]: I removed agentlogs and added screenshots/* as suggested by @hundao ,not jus removing spaces
